aws-rds-dbclusterendpoint/src/main/java/software/amazon/rds/dbclusterendpoint/BaseHandlerStd.java

```diff
@@ -0,0 +1,148 @@
+            final ProxyClient<RdsClient> proxyClient,
+            final ProgressEvent<ResourceModel, CallbackContext> progress,
+            final Tagging.TagSet previousTags,
+            final Tagging.TagSet desiredTags
+    ) {
+        return proxy.initiate("rds::tag-db-cluster-endpoint", proxyClient, progress.getResourceModel(), progress.getCallbackContext())
+                .translateToServiceRequest(Translator::describeDbClustersEndpointRequest)
+                .backoffDelay(config.getBackoff())
+                .makeServiceCall((describeRequest, proxyInvocation) -> proxyInvocation.injectCredentialsAndInvokeV2(
+                        describeRequest,
+                        proxyInvocation.client()::describeDBClusterEndpoints
+                )).handleError((describeRequest, exception, client, resourceModel, ctx) -> Commons.handleException(
+                        ProgressEvent.progress(resourceModel, ctx),
+                        exception,
+                        DEFAULT_DB_CLUSTER_ENDPOINT_ERROR_RULE_SET
+                ))
+                .done((describeRequest, describeResponse, invocation, resourceModel, ctx) -> {
+                    final Tagging.TagSet tagsToAdd = Tagging.exclude(desiredTags, previousTags);
+                    final Tagging.TagSet tagsToRemove = Tagging.exclude(previousTags, desiredTags);
+
+                    if (tagsToAdd.isEmpty() && tagsToRemove.isEmpty()) {
+                        return progress;
+                    }
+
+                    final String arn = describeResponse.dbClusterEndpoints().stream().findFirst().get().dbClusterEndpointArn();
+
+                    try {
+                        Tagging.removeTags(proxyClient, arn, Tagging.translateTagsToSdk(tagsToRemove));
+                        Tagging.addTags(proxyClient, arn, Tagging.translateTagsToSdk(tagsToAdd));
+                    } catch (Exception exception) {
+                        return Commons.handleException(
+                                progress,
+                                exception,
+                                Tagging.bestEffortErrorRuleSet(tagsToAdd, tagsToRemove).orElse(DEFAULT_DB_CLUSTER_ENDPOINT_ERROR_RULE_SET)
+                        );
+                    }
+                    return progress;
+                });
+
+    }
+
+    protected DBClusterEndpoint fetchDBClusterEndpoint(
+            final ResourceModel model,
+            final ProxyClient<RdsClient> proxyClient
+    ) {
+        final DescribeDbClusterEndpointsResponse response = proxyClient.injectCredentialsAndInvokeV2(
+                Translator.describeDbClustersEndpointRequest(model),
+                proxyClient.client()::describeDBClusterEndpoints
+        );
+
+        final Optional<DBClusterEndpoint> clusterEndpoint = response
+                .dbClusterEndpoints().stream().findFirst();
+
+        return clusterEndpoint.orElseThrow(() -> new CfnNotFoundException(ResourceModel.TYPE_NAME,
```

Comment:
Unfortunately, RDS SDK already defines `DBClusterEndpointNotFoundException` and throws it as expected on `ModifyDBClusterEndpoint` requests for non-existent Cluster Endpoints. I believe we should not name our own exception `DBClusterEndpointNotFoundException` since it will be very easy to use wrong type accidentally. We can create our own exception, but the naming will be standing out from the rest of exceptions we have.

Alternatively, we could use SDK exception.